rllib/core/models/torch/encoder.py

```diff
@@ -148,91 +158,89 @@ def _forward(self, inputs: NestedDict, **kwargs) -> NestedDict:
+                        "b, l, h",
+                        h=self.config.hidden_dim,
+                        l=self.config.num_lstm_layers,
                     ),
                     "c": TorchTensorSpec(
-                        "b, l, h", h=self.config.hidden_dim, l=self.config.num_layers
+                        "b, l, h",
+                        h=self.config.hidden_dim,
+                        l=self.config.num_lstm_layers,
                     ),
                 },
-                SampleBatch.SEQ_LENS: None,
             }
         )
 
     @override(Model)
     def get_output_specs(self) -> Optional[Spec]:
         return SpecDict(
             {
-                ENCODER_OUT: TorchTensorSpec("bxt, h", h=self.config.output_dims[0]),
+                ENCODER_OUT: TorchTensorSpec("b, t, d", d=self.config.output_dims[0]),
                 STATE_OUT: {
                     "h": TorchTensorSpec(
-                        "b, l, h", h=self.config.hidden_dim, l=self.config.num_layers
+                        "b, l, h",
+                        h=self.config.hidden_dim,
+                        l=self.config.num_lstm_layers,
                     ),
                     "c": TorchTensorSpec(
-                        "b, l, h", h=self.config.hidden_dim, l=self.config.num_layers
+                        "b, l, h",
+                        h=self.config.hidden_dim,
+                        l=self.config.num_lstm_layers,
                     ),
                 },
             }
         )
 
     @override(Model)
     def get_initial_state(self):
-        config = self.config
         return {
-            "h": torch.zeros(config.num_layers, config.hidden_dim),
-            "c": torch.zeros(config.num_layers, config.hidden_dim),
+            "h": torch.zeros(self.config.num_layers, self.config.hidden_dim),
+            "c": torch.zeros(self.config.num_layers, self.config.hidden_dim),
         }
 
     @override(Model)
     def _forward(self, inputs: NestedDict, **kwargs) -> NestedDict:
-        x = inputs[SampleBatch.OBS].float()
-        states = inputs[STATE_IN]
-        # states are batch-first when coming in
-        states = tree.map_structure(lambda x: x.transpose(0, 1), states)
-
-        x = add_time_dimension(
-            x,
-            seq_lens=inputs[SampleBatch.SEQ_LENS],
-            framework="torch",
-            time_major=not self.config.batch_first,
-        )
-        states_o = {}
-        x, (states_o["h"], states_o["c"]) = self.lstm(x, (states["h"], states["c"]))
-
-        x = self.linear(x)
-        x = x.view(-1, x.shape[-1])
-
-        return {
-            ENCODER_OUT: x,
-            STATE_OUT: tree.map_structure(lambda x: x.transpose(0, 1), states_o),
-        }
+        out = inputs[SampleBatch.OBS].float()
 
+        # States are batch-first when coming in. Make them layers-first.
+        states_in = tree.map_structure(lambda s: s.transpose(0, 1), inputs[STATE_IN])
```

Comment:
I think the torch code looks cleaner here. Can me match tf to this?

Reply:
done, removed the list(range(2, .. ))) stuff for TF, as it's probably not needed (state tensors are always 1D).